### ISSUE TITLE
Standardise the search and fetch APIs of all subclasses of BaseClient

### ISSUE THREAD

**Issue by Cadair** [sunpy/sunpy#4895]
Currently the signatures in the base client are:

```python
    def search(self, *args, **kwargs):

    def fetch(self, query_results, *, path=None, overwrite=False, progress=True,
              max_conn=5, downloader=None, wait=True, **kwargs):
```

A lot of the clients we have make up their own signature, we should decide on a standard signature in the base client and then try and unify the subclasses as much as possible. I propose the following signature for search:

```python
    def search(self, *query, **kwargs):
```

fetch is more complex, there is part of me which thinks a very minimal signature like this:
```
    def fetch(self, *query_results, path=None, downloader=None, wait=True, **kwargs):
```
would be fine and users of individual clients (not Fido) can pass the `overwrite`, `progress` and `max_conn` arguments to the `Downloader()` constructor.

Another option would be to do something like:

```python
     def fetch(self, *query_results, path=None, downloader=None, downloader_kwargs=None, wait=True, **kwargs):
```

Finally, it's important that all clients allow `**kwargs` in both fetch and search so Fido can reliably pass through all kwargs to all clients.

**Comment by exitflynn** [comment]
i agree with this and prefer the first version of the fetch signature, the second one seems a teeny bit more limiting. has there been any further discussion on this?
